### ISSUE TITLE
Suspense and ErrorBoundary Doesn't render Ok() case if initial render is Err() until transitioning from Err -> Ok -> Err -> Ok

### ISSUE THREAD

**Issue by zoomiti** [leptos-rs/leptos#2508]
**Describe the bug**
Basically if an initial page render renders a suspense with an err, the error boundary fallback is rendered. However if through signals the resource that is suspended succeeds on a refetch the success case is not rendered. This is fixed if one more transition to err and then back to ok is performed. This might make more sense by just following the reproduction instructions.

**Leptos Dependencies**
```toml
leptos = { version = "0.6", features = ["nightly"] }
leptos_axum = { version = "0.6", optional = true }
leptos_meta = { version = "0.6", features = ["nightly"] }
leptos_router = { version = "0.6", features = ["nightly"] }
```

**To Reproduce**
Using the [fallible resource branch of my min repro repo](https://github.com/zoomiti/min-repro/tree/fallible_resource)
1. Navigate to the page with query id=6,
2. Input 4 into the text box that will update the query signal and refetch the resource
3. Nothing will be rendered at first
4. Input 6 to render the Error case again
5. Input 4 and the Ok case will finally be rendered

**Expected behavior**
If an initial page load renders the fallback of an error boundary, if a signal refetches the resources into an ok, the ok branch should be rendered immediately.

**Screenshots**
N/A hopefully but happy to provide if needed

**Additional context**
This uses a server fn as the resource fetcher, to mirror my actual use case. However I can reproduce the bug without it being a server fn.

**Comment by gbj** [comment]
I can reproduce this, and it seems to be an issue with the Suspense not tracking the resource correctly if it's in the fallback branch of the ErrorBoundary.

Adding a simple `.track()` for the resource outside the ErrorBoundary resolves the issue (and causes some warnings that are not a problem)

```rust
<Suspense>
    {resource.track()}
    <ErrorBoundary fallback=|_| view! { "Error" }>{resource}</ErrorBoundary>
</Suspense>
```

Ultimately it's a non-obvious hydration issue here that I'll have to look into more at some point, or which will be superseded by changes in 0.7.

**Comment by zoomiti** [comment]
I appreciate at least having a patch fix for my app in the meantime. I'll use the track technique for now. Thank you!

**Comment by gbj** [comment]
I have tested and I believe this is fixed on 0.7/git main, but feel free to open a new issue if that is not the case.
